dev/tools/push-images

```diff
@@ -83,61 +82,47 @@ def build_and_push_image(args, service_name, srcdir, dockerfile_path):
     if args.kind_cluster_name:
         load_kind_image(args.kind_cluster_name, service_name, srcdir)
 
-# This only builds and pushes images in directories with
-# 1) go.mod file
-# 2) images/*/Dockerfile
-def push_image_for_go_mod(args, mod_dir):
-    """Builds and pushes docker images."""
-    images_dir = os.path.join(mod_dir, "images")
-
-    if not os.path.isdir(images_dir):
-        return
+def main(args):
+    """Builds and pushes all discovered Docker images."""
+    excluded_dirs = {
+        ".git",
+        ".venv",
+        "myenv",
+        ".pytest_cache",
+        "build",
+        "dist",
+        "bin",
+        ".vscode",
+        "__pycache__",
+        "site",
+        "k8s",
+        "dev",
+        "docs"
+    }
+
+    for root, dirs, files in os.walk(".", topdown=True):
+        # Exclude directories
+        dirs[:] = [d for d in dirs if d not in excluded_dirs]
+
+        if "Dockerfile" in files:
+            dockerfile_path = os.path.join(root, "Dockerfile")
+            context_dir = root
+            service_name = os.path.basename(root)
```

Comment:
We need to handle the case when there are dup folder names. This can be handled later. Please add a TODO.